packages/dialog/test/dialog.test.js

```diff
@@ -161,4 +162,28 @@ describe('vaadin-dialog', () => {
       expect(contentMinWidth).to.be.gt(0);
     });
   });
+
+  describe('focus restoration', () => {
+    let dialog, button;
+
+    beforeEach(() => {
+      dialog = fixtureSync('<vaadin-dialog></vaadin-dialog>');
+      button = fixtureSync('<button></button>');
+      button.focus();
+    });
+
+    it('should move focus to dialog on open', async () => {
+      dialog.opened = true;
+      await oneEvent(dialog.$.overlay, 'vaadin-overlay-open');
+      expect(isElementFocused(dialog.$.overlay)).to.be.true;
+    });
+
+    it('should restore focus on dialog close', async () => {
```

Comment:
Tests could be extended to also cover disconnecting `vaadin-dialog` from the DOM.

Reply:
Note, there is already a test to ensure the overlay is closed on dialog detach:

https://github.com/vaadin/web-components/blob/731706a61dce7640e286c7a64fc67c4daed29161/packages/dialog/test/dialog.test.js#L116-L119

My idea was to only add a basic "smoke" test that would ensure `restoreFocusOnClose` is enabled.